renderer/features/wallet/walletSlice.js

```diff
@@ -232,9 +232,10 @@ export const connectWallet = walletId => {
       dispatch(fetchBalance());
     });
 
-    const handleChannelEventData = proxyValue(channelEvent =>
-      dispatch(receiveChannelEvent(channelEvent))
-    );
+    const handleChannelEventData = proxyValue(channelEvent => {
+      dispatch(receiveChannelEvent(channelEvent));
+      dispatch(fetchBalance());
+    });
```

Comment:
let's move the fetchBalance into receiveChannelEvent -- that's meant to be a singular place that everything happens when we receive said event.

Reply:
I mean, we would import the fetchBalance from walletSlice in the channelSlice, and import the receiveChannelEvent from the channelSlice in the walletSlice.

The problem is maybe the fact that we're subscribing here in walletSlice to these events from all over the app. It seems to me like the right "time" to subscribe, cause we are connecting to the wallet, but not right place. But I'm still confused by this react / redux paradigm as I'm new to it.

Reply:
yeah true, hm.  I guess leave it for now but we should figure out how to better structure to avoid dependency cycle.